guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/ConfiguredRepositories.java

```diff
@@ -0,0 +1,196 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.structure.backend.repositories;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.guvnor.structure.config.SystemRepositoryChangedEvent;
+import org.guvnor.structure.repositories.NewBranchEvent;
+import org.guvnor.structure.repositories.Repository;
+import org.guvnor.structure.repositories.impl.git.GitRepository;
+import org.guvnor.structure.server.config.ConfigGroup;
+import org.guvnor.structure.server.config.ConfigurationService;
+import org.guvnor.structure.server.repositories.RepositoryFactory;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.java.nio.file.FileSystem;
+
+import static org.guvnor.structure.server.config.ConfigType.*;
+import static org.uberfire.backend.server.util.Paths.*;
+
+/**
+ * Cache for configured repositories.
+ */
+@ApplicationScoped
+public class ConfiguredRepositories {
+
+    private ConfigurationService configurationService;
+    private RepositoryFactory    repositoryFactory;
+    private Repository           systemRepository;
+
+    private Map<String, Repository> repositoriesByAlias      = new HashMap<>();
+    private Map<Path, Repository>   repositoriesByBranchRoot = new HashMap<>();
+
+    public ConfiguredRepositories() {
+    }
+
+    @Inject
+    public ConfiguredRepositories( final ConfigurationService configurationService,
+                                   final RepositoryFactory repositoryFactory,
+                                   final @Named( "system" ) Repository systemRepository ) {
+        this.configurationService = configurationService;
+        this.repositoryFactory = repositoryFactory;
+        this.systemRepository = systemRepository;
+    }
+
+    @SuppressWarnings( "unchecked" )
+    @PostConstruct
+    public void loadRepositories() {
+        repositoriesByAlias.clear();
+        repositoriesByBranchRoot.clear();
+
+        final List<ConfigGroup> repoConfigs = configurationService.getConfiguration( REPOSITORY );
+        if ( !(repoConfigs == null || repoConfigs.isEmpty()) ) {
+            for ( final ConfigGroup configGroup : repoConfigs ) {
+                final Repository repository = repositoryFactory.newRepository( configGroup );
+
+                add( repository );
+
+            }
+        }
+    }
+
+    /**
+     *
+     * @param alias Name of the repository.
+     * @return Repository that has a random branch as a root, usually master if master exists.
+     */
+    public Repository getRepositoryByRepositoryAlias( final String alias ) {
+        return repositoriesByAlias.get( alias );
+    }
+
+    /**
+     * This can also return file system repository.
```

Comment:
Comment is a little confusing "..File System Repository" Would "...also return System Repository" be clearer?